kernel/src/vtpm/ptp/crb.rs

```diff
@@ -268,6 +311,51 @@ impl Tpm {
         self.reset()
     }
 
+    pub fn extend_svsm_version(&mut self) -> Result<()> {
+        let svsm_version_digests = if let Some(digests) = vtpm_get_svsm_version() {
+            digests
+        } else {
+            return Err(Error::ExtendSvsm);
+        };
+        let mut tpm2_command = [0u8; MAX_TPM_PCR_EXTEND_CMD_SIZE];
+        let mut tpm_cmd_size =
+            if let Ok(cmd_size) = tpm2_pcr_extend(&svsm_version_digests, 0, &mut tpm2_command) {
+                cmd_size
+            } else {
+                return Err(Error::ExtendSvsm);
+            };
+
+        let backend = vtpm_get_locked();
+
+        // Send TPM_STARTUP CMD
+        let mut tpm_startup_cmd = TPM_STARTUP_CMD;
+        if backend
+            .send_tpm_command(&mut tpm_startup_cmd, &mut TPM_STARTUP_CMD.len(), 0)
+            .is_ok()
+        {
+            log::info!("send pcr extend request success\n");
+            self.request_completed(true);
+        } else {
+            log::error!("send pcr extend request fail!\n");
+            self.request_completed(false);
+            return Err(Error::ExtendSvsm);
+        }
+
+        // Send PCR_EXTEND CMD
+        if backend
+            .send_tpm_command(&mut tpm2_command[..], &mut tpm_cmd_size, 0)
+            .is_ok()
+        {
+            log::info!("send pcr extend request success\n");
+            self.request_completed(true);
```

Comment:
I suppose the `request_complete` is not required because the code does not touch the CRB status. Also, please consider move the `extend_svsm_version` out from the vtpm module.

Reply:
Moved to tdx/measurement.rs